### PR TITLE
fix: timestamp query expansion

### PR DESCRIPTION
**Pull request by axejintao**

### CODE REVIEW

File: src/aws/dynamodb.utils.ts [hunk 1]
```diff
@@ -28,3 +30,7 @@ export function getLeaderboardKey(chain: Chain): string {
 export function getChainStartBlockKey(chain: Chain, block: number): string {
   return `${chain.network}_${block}`;
 }
+
+export function getVaultEntityId({ network }: Chainish, { address }: Vaultish): string {
```
Comment (axejintao) [inline src/aws/dynamodb.utils.ts:34]: we should start to transition to using this for `id` for all vault entities, which currently we base off only address

File: src/vaults/vaults.service.ts [hunk 2]
```diff
@@ -115,48 +115,73 @@ export class VaultsService {
...
-      if (snapshotItem) {
-        if (!snapshotItem.pricePerFullShare && snapshotItem.ratio) {
-          snapshotItem.pricePerFullShare = snapshotItem.ratio;
+      // from smaller to greater, we need this order for search to work correctly
+      const preparedVaultsChartData = vaultChartData.reverse();
+      const remainingTimestamps = [];
+
+      // iterate over timestamps updating the map as snapshots are found
+      for (const timestamp of sortedTimestamps) {
+        const snapshotItem = preparedVaultsChartData.find((i) => Number(i.timestamp) >= timestamp);
```
Comment (mrbasado) [inline src/vaults/vaults.service.ts:169]: does this always find just 1 tiemstamp?

Comment (axejintao) [reply src/vaults/vaults.service.ts:169]: yes that is what find does, undefined or one

Comment (mrbasado) [reply src/vaults/vaults.service.ts:169]: i see comment now on required presort of that list

**Review by mrbasado — APPROVED** [review]

**Pull request merged**